packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,230 @@
+        )
+        t_col.insert_one(new_team.model_dump())
+
+        return new_team
+
+    @classmethod
+    def add_team_participant(cls, team_name: str, user_id: str) -> List[str] | None:
+        team = cls.fetch_team(team_name=team_name)
+        if not team:
+            return None
+
+        team = HackathonTeam(
+            team_name=team_name, team_members=team["team_members"], team_type=team["team_type"],
+        )
+        team.team_members.append(user_id)
+
+        cls.update_team_fields(team)
+
+        return team.team_members
+
+    @classmethod
+    def fetch_team(cls, team_name: str = "", team_id: str = "") -> Dict[str, Any]:
+        team: Dict[str, Any] = t_col.find_one(filter={"team_name": team_name})
+
+        if team_id:
+            team = t_col.find_one(filter={"_id": ObjectId(team_id)})
+            return team
+
+        return team
+
+    @classmethod
+    def fetch_teams(cls) -> JSONResponse:
+        teams = list(t_col.find())
+
+        if not teams:
+            return JSONResponse(content={"message": "No teams were found in db"}, status_code=404)
+
+        return JSONResponse(content={"teams": json.loads(dumps(teams))}, status_code=200)
+
+    @classmethod
+    def get_team(cls, object_id: str) -> JSONResponse:
+        specified_team = cls.fetch_team(team_id=object_id)
+
+        if not specified_team:
+            return JSONResponse(content={"message": "The team was not found"}, status_code=404)
+
+        return JSONResponse(content={"participant": json.loads(dumps(specified_team))}, status_code=200)
+
+    @staticmethod
+    def delete_team(object_id: str) -> JSONResponse:
+        delete_team = t_col.find_one_and_delete(
+            filter={"_id": ObjectId(object_id)},
+        )
+
+        if not delete_team:
+            return JSONResponse(content={"message": "The team was not found"}, status_code=404)
+
+        return JSONResponse(content={"message": json.loads(dumps(delete_team))}, status_code=200)
+
+    @staticmethod
+    def team_count() -> JSONResponse:
+        count = t_col.count_documents({})
+
+        if not count:
+            return JSONResponse(content={"message": "No teams were found"}, status_code=404)
+
+        return JSONResponse(content={"teams": count})
+
+    @classmethod
+    def update_team(cls, object_id: str, update_table_model: UpdateTeam) -> JSONResponse:
+        team = cls.fetch_team(team_id=object_id)
+
+        if not team:
+            return JSONResponse(content={"message": "The team was not found"}, status_code=404)
+
+        result = cls.update_team_fields(
+            update_table_model, object_id=object_id,
+        )
+
+        return JSONResponse(content={"teams": json.loads(dumps(result))}, status_code=200)
+
+    @classmethod
+    def move_team_members(cls, move_members_model: MoveTeamMembers) -> JSONResponse:
+        move_members_dump: Dict[str, Any] = move_members_model.model_dump()
+        from_team: str = move_members_dump["from_team"]
```

Comment:
Why access the values like this instead of with the dot operator?

Reply:
I don't know how I've missed that but that's pretty huge. Why serialize data when you can directly access it within the provisioned object?